Add tests for premium operators

jhj2713 opened the pull request:
Tests for aggregation operators : TimeSinceLastFalse, TimeSinceLastMax, TimeSinceLastMin, TimeSinceLastTrue, Variance 
   
Tests for transform operators : PhoneNumberToCountry, PolarityScore, PunctuationCount, Quarter, SavgolFilter, Season, Sign, StopwordCount, SubRegionCodeToRegion, TitleWordCount, UpperCaseCount, UpperCaseWordCount, URLToProtocol, ZIPCodeToState

jhj2713 committed:
Add tests for aggregation operators

Add tests for : TimeSinceLastFalse, TimeSinceLastMax, TimeSinceLastMin,
TimeSinceLastTrue, Variance

jhj2713 committed:
Add tests for transform operators

Add tests for : PhoneNumberToCountry, PolarityScore, PunctuationCount,
Quarter, SavgolFilter, Season, Sign, StopwordCount,
SubRegionCodeToRegion, TitleWordCount, UpperCaseCount,
UpperCaseWordCount, URLToProtocol, ZIPCodeToState

cezanne commented:
Please check.. 
```
cezanne@cezdev2:~/work/qufafeat/featuretools/tests$ pytest-3 primitive_tests/test_premium_aggregation_primitive.py -v
============================================= test session starts =============================================
platform linux -- Python 3.8.10, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/cezanne/work/qufafeat, inifile: setup.cfg
plugins: xdist-1.26.1, cov-2.6.1, forked-1.3.0
collected 7 items                                                                                             

primitive_tests/test_premium_aggregation_primitive.py::test_AutoCorrelation PASSED                      [ 14%]
primitive_tests/test_premium_aggregation_primitive.py::test_Correlation PASSED                          [ 28%]
primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastFalse PASSED                   [ 42%]
primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastMax PASSED                     [ 57%]
primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastMin PASSED                     [ 71%]
primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastTrue PASSED                    [ 85%]
primitive_tests/test_premium_aggregation_primitive.py::test_Variance PASSED                             [100%]

============================================== warnings summary ===============================================
featuretools/tests/primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastMax
  /usr/lib/python3/dist-packages/numpy/core/fromnumeric.py:61: FutureWarning: 
  The current behaviour of 'Series.argmax' is deprecated, use 'idxmax'
  instead.
  The behavior of 'argmax' will be corrected to return the positional
  maximum in the future. For now, use 'series.values.argmax' or
  'np.argmax(np.array(values))' to get the position of the maximum
  row.
    return bound(*args, **kwds)

featuretools/tests/primitive_tests/test_premium_aggregation_primitive.py::test_TimeSinceLastMin
  /usr/lib/python3/dist-packages/numpy/core/fromnumeric.py:61: FutureWarning: 
  The current behaviour of 'Series.argmin' is deprecated, use 'idxmin'
  instead.
  The behavior of 'argmin' will be corrected to return the positional
  minimum in the future. For now, use 'series.values.argmin' or
  'np.argmin(np.array(values))' to get the position of the minimum
  row.
    return bound(*args, **kwds)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================== 7 passed, 2 warnings in 1.90 seconds =====================================
```

cezanne commented:
`test_premium_transform_primitive.py()` has produced too many errors.
Please check [this attachment](https://github.com/oslab-ewha/qufafeat/files/6838171/polarity_score.txt), which is a test output for running test_PolarityScore.

cezanne commented:
transform test의 warning은 아래와 같이 수정을 하면 오류가 나지 않습니다. 기존 방식은 deprecated된 API라서 warning이 발생한 거 같습니다.

```
diff --git a/featuretools/primitives/premium/aggregation_primitives.py b/featuretools/primitives/premium/aggregation_primitives.py
index 311a99db..d27c2517 100644
--- a/featuretools/primitives/premium/aggregation_primitives.py
+++ b/featuretools/primitives/premium/aggregation_primitives.py
@@ -137,7 +137,7 @@ class TimeSinceLastMax(AggregationPrimitive):
 
     def get_function(self):
         def time_since_last_max(values, arr, time=None):
-            maxIndex = np.argmax(arr)
+            maxIndex = arr.idxmax()
             time_since = time - values.iloc[maxIndex]
             return time_since.total_seconds()
 
@@ -167,7 +167,7 @@ class TimeSinceLastMin(AggregationPrimitive):
 
     def get_function(self):
         def time_since_last_min(values, arr, time=None):
-            minIndex = np.argmin(arr)
+            minIndex = arr.idxmin()
             time_since = time - values.iloc[minIndex]
             return time_since.total_seconds()
 
```

cezanne commented:
transform 테스트에서 발생하던 오류는 nltk 패키지 설치 후, 몇가지 data 파일을 설치해 줘야 하는 거 같네요.

```
>>> import nltk
>>> nltk.download('vader_lexicon')
>>> nltk.download('stopwords')
```

jhj2713 committed:
Add nltk.download and revise deprecated API

Add nltk.download() in __init__.py
Revise deprecated API in aggregation operator
